### PR TITLE
Remove comments from mir-opt MIR dumps

### PR DESCRIPTION
**Pull request by saethlin**
See https://rust-lang.zulipchat.com/#narrow/stream/189540-t-compiler.2Fwg-mir-opt/topic/Line.20numbers.20in.20mir-opt.20tests/near/363849874

In https://github.com/rust-lang/rust/pull/99780 there is mention that "there has been a zulip conversation about disabling line numbers with mixed opinions" which to me means that some people opposed this. I can't find the referenced conversation so... here we go.

The current situation is quite chaotic. It's not hard to find MIR diffs which contain

* Absolute line numbers
* Relative line numbers
* Substituted line numbers (LL)
For example: https://github.com/rust-lang/rust/blob/408bbd040613f6776e0a7d05d582c81f4ddc189a/tests/mir-opt/inline/inline_shims.drop.Inline.diff#L10-L17

And sometimes adding a comment at the top of a mir-opt test generates a diff in the test because a line number changed: https://github.com/rust-lang/rust/pull/98112/files#diff-b8cf4bcce95078e6a3faf075e9abf6864872fb28a64d95c04f04513b9e3bbd81

And irrelevant changes to the standard library can generate diffs in mir-opt tests: https://github.com/rust-lang/rust/pull/110694/files#diff-bf96b0e7c67b8b272814536888fd9428c314991e155beae1f0a2a67f0ac47b2c
https://github.com/rust-lang/rust/commit/769886cc35ce08b76839f4cf72b8af1161c432e1

I think we should, specifically in mir-opt tests, completely remove the comments, or insert placeholders for all line and column numbers.

### CODE REVIEW

File: tests/mir-opt/building/custom/consts.statics.built.after.mir [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-        _1 = const {alloc1: &i32};       // scope 0 at $SRC_DIR/core/src/intrinsics/mir.rs:LL:COL
+        _1 = const {alloc1: &i32};
                                          // mir::Constant
                                          // + span: $DIR/consts.rs:27:31: 27:32
                                          // + literal: Const { ty: &i32, val: Value(Scalar(alloc1)) }
-        _2 = const {alloc2: *mut i32};   // scope 0 at $SRC_DIR/core/src/intrinsics/mir.rs:LL:COL
+        _2 = const {alloc2: *mut i32};
                                          // mir::Constant
                                          // + span: $DIR/consts.rs:28:38: 28:39
                                          // + literal: Const { ty: *mut i32, val: Value(Scalar(alloc2)) }
```
Comment (bjorn3) [inline tests/mir-opt/building/custom/consts.statics.built.after.mir:16]: Can you put the `// mir::Constant` on the same line as the `_2 = const {alloc2: *mut i32};`?

**Comment by saethlin** [comment]
r? mir-opt

**Comment by scottmcm** [comment]
> blessing with `debug = true` is still not effective.

Not this PR, but maybe emitting a warning for that could make sense?

(And I assume it's `debug-std = true` that's the major issue?  `debug = true; debug-std = false` should be fine for blessing these?)

**Comment by saethlin** [comment]
The problem is that the tests don't get blessed _at all_ because they are `// ignore-debug`.

Yes, it's the standard library code that's the problem, and I think this can be gracefully solved by piggybacking on the work Pietro did.

**Comment by lqd** [comment]
It seems this PR removed comments from `-Zdump-mir` dumps that were not mir-opt tests. I'll open a PR.

**Comment by shepmaster** [comment]
This also removed functionality from the playground. I'll delete that code now.

**Pull request merged**